drafts/25-xxx-specifications.txt

```diff
@@ -0,0 +1,491 @@
+      on directive lines.
+
+ws03. Whitespace characters are significant in determining token
+      boundaries in preprocessor directive lines.
+
+ws02. Whitespace characters are significant in determining token
+      boundaries in fixed-form and free-fort Fortran source fragments.
+
+ws03. Whitespace characters are significant in determining token
+      boundaries in fixed-form and free-fort Fortran comment lines.
+
+ws04. In fixed-form input, macro names are not recognized as such when
+      spaces are inserted into their invocations.
+
+
+2.2 Comments
+------------
+
+co01. Directive lines may contain C-style '/*' ... '*/' comments.
+
+co02. Directive lines shall not contain C style '//' comments.
+
+co02. '/*' ... '*/' comments on directive lines shall extend past a
+      new-line only if the line ends in '\' new-line, indicating a
+      continuation line. (Maybe controversial)
+
+co03. '/*' ... '*/' comments on directive lines are replaced by a
+      single space, as specified in C 2023 5.1.1.2 bullet 3.
+
+co04. Directive lines shall not contain Fortran '!' style comments.
+
+co05. Directive lines (by definition) cannot contain Fortran
+      fixed-form 'C' or '*' style comments.
+
+
+2.4 Token lexicon
+-----------------
+[gak: List the character sequences that are recognized as separate tokens
+on directives and in Fortran source fragments. This should include
+Fortran identifiers, Fortran numeric constants, Fortran character
+constants, the Fortran operators, and the C operators used to construct
+constant-integer-expression.]
+
+
+
+3 #-Directives
+==============
+
+di01. The '#define' macro directive
+
+di02. The '#define' function-like macro directive
+
+di03. The '#undef' directive
+
+di04. The '#include' directive
+
+di03. The '#if', '#ifdef', '#ifndef', '#elif', '#elifdef',
+      '#elifndef', '#else', '#endif' directive collective
+
+di04. The '#error' and '#warning' directives
+
+di05. The '#line' directive
+
+di06. The '#pragma' directive
+
+di07. The non-directive directive
+
+
+
+
+3.1 The '#define' macro directive
+---------------------------------
+
+
+3.2 The '#define' function-like macro directive
+-----------------------------------------------
+
+
+3.3 The '#undef' directive
+--------------------------
+
+
+
+3.4 The '#include' directive
+----------------------------
+
+
+3.3 The '#if', '#ifdef', '#ifndef', '#elif', '#elifdef',
+      '#elifndef', '#else', '#endif' directive collective
+---------------------------------------------------------
+
+
+3.4 The '#error' and '#warning' directives
+------------------------------------------
+
+
+3.5 The '#line' directive
+-------------------------
+
+
+3.6 The '#pragma' directive
+---------------------------
+
+
+3.7 The non-directive directive
+-------------------------------
+
+
+
+
+4 Macro recognition and expansion
+---------------------------------
+
+
+
+4.8 The 'defined' operator
+--------------------------
```

Comment:
The `defined` operator is only valid in `#if and #elif directives`, so it probably belongs in that section (currently numbered 5).

Reply:
```suggestion
4 Macro recognition and expansion
---------------------------------


```